pkg/frontend/querymiddleware/querysharding.go

```diff
@@ -238,7 +238,12 @@ func newQuery(ctx context.Context, r MetricsQueryRequest, engine *promql.Engine,
 			r.GetQuery(),
 			util.TimeFromMillis(r.GetStart()).Add(1*time.Nanosecond),
 			util.TimeFromMillis(r.GetEnd()),
-			time.Duration(r.GetStep())*time.Millisecond,
+			// TODO(56quarters): Need to actually figure out what the Prometheus engine
```

Comment:
Why do we need to handle remote read requests here at all? As far as I can see, this method is only used for the sharding and subquery spin-off middlewares, which should only be used for range and instant queries.